src/physics/physics_model.cpp

```diff
@@ -127,12 +159,30 @@ ::convective_numerical_split_flux(const std::array<real,nstate> &conservative_so
                                   const std::array<real,nstate> &conservative_soln2) const
 {
     // TO DO: Update for when nstate > nstate_baseline_physics
+    //std::array<dealii::Tensor<1,dim,real>,nstate> conv_num_split_flux;
+    //if constexpr(nstate==nstate_baseline_physics) {
+    //    conv_num_split_flux = physics_baseline->convective_numerical_split_flux(conservative_soln1,conservative_soln2);
+    //} else {
+    //    // TO DO, make use of the physics_model object for nstate>nstate_baseline_physics
+    //    std::abort();
+    //}
+
     std::array<dealii::Tensor<1,dim,real>,nstate> conv_num_split_flux;
-    if(nstate==nstate_baseline_physics) {
+    if constexpr(nstate==nstate_baseline_physics) {
```

Comment:
I think the condition should actually be `if (model->has_additional_convective_flux_terms)` would be more appropriate than `if(nstate==nstate_baseline_physics)`; this way we're not eliminating the possibility that the model has additional terms to the existing convective numerical flux in the `istate <= nstate_baseline_physics`. 

The `bool has_additional_convective_flux_terms` should be introduced in the `ModelBase` class as an input to the contructor; would also be a good idea to similarly do this for the dissipative flux, source term, and physical source term, by similarly introducing `bool has_additional_[name]_terms`

Reply:
We could simply open an issue for this if you feel like it's out of scope

Reply:
Ok if you do, open an issue and provide a link to this conversation